src/main.ts

```diff
@@ -0,0 +1,656 @@
+
+db.query(
+  `create table if not exists "tokens" ( user_id text primary key, refresh_token text, expiration_date number, user_name string );`
+).all();
+
+app.get("/", (c) => {
+  const userID = c.req.query("user_id") || `u_${nanoid()}`;
+
+  return c.html(`
+		<body>
+			<p> Your user id is ${userID} </p>
+			<a href="/auth/github/authorize?user_id=${userID}"> Link GitHub </a>
+      &nbsp
+			<a href="/tokens?user_id=${userID}"> See my tokens </a>
+      &nbsp
+      <a href="/commits?user_id=${userID}"> See my commits </a>
+      &nbsp
+      <a href="/PRs?user_id=${userID}"> See my PRs </a>
+      &nbsp
+      <a href="/summary?user_id=${userID}"> Summary of what I did today! </a>
+		</body>
+		`);
+});
+
+app.get("/auth/github/authorize", async (c) => {
+  const userID = c.req.query("user_id");
+
+  if (!userID) {
+    return c.json("Unauthorized", 401);
+  }
+  const { state, url } = await authz.github.authorize();
+
+  const secure = c.req.url.startsWith("https://");
+
+  const value = pipe(state, JSON.stringify, stringToBase64);
+  setCookie(c, "state", value, {
+    maxAge: 10 * seconds.minute,
+    httpOnly: true,
+    ...(secure ? { secure: secure, sameSite: "none" } : {}),
+  });
+  const valueUser = pipe(userID, JSON.stringify, stringToBase64);
+  setCookie(c, "user", valueUser, {
+    maxAge: 10 * seconds.minute,
+    httpOnly: true,
+    ...(secure ? { secure: secure, sameSite: "none" } : {}),
+  });
+
+  return c.redirect(url.toString());
+});
+
+app.get("/auth/github/callback", async (c) => {
+  const cookie = getCookie(c, "state");
+  const userID = getCookie(c, "user");
+
+  if (!userID) {
+    return c.json({ message: "unauthorized" }, { status: 401 });
+  }
+
+  if (!cookie) {
+    return c.json({ message: "No state" }, { status: 400 });
+  }
+
+  const state = pipe(cookie, base64ToString, JSON.parse);
+  const response = await authz.github.callback(
+    new URL(c.req.url),
+    state,
+    GITHUB_CALLBACK.toString()
+  );
+
+  const currentTimestamp = Date.now();
+
+  const endDate =
+    currentTimestamp + (response.refresh_token_expires_in as number) * 1000;
+
+  const octokit = new Octokit({ auth: response.access_token });
+  const userName = (await octokit.request("GET /user")).data.login;
+
+  // encrypt the refresh token
+  const bytes = await encrypt(
+    new TextEncoder().encode(response.refresh_token),
+    key
+  );
+  const buffer = Buffer.from(bytes);
+  const base64String = buffer.toString("base64");
+
+  // Save userId, refresh token, expiration date of the refresh token and the user name to db
+  db.query(
+    "insert into tokens (user_id, refresh_token, expiration_date, user_name) values ($userId, $refreshToken, $endDate, $userName)"
+  ).all({
+    $userId: userID!,
+    $refreshToken: base64String,
+    $endDate: endDate,
+    $userName: userName,
+  });
+
+  // save the userId, access token and the expiration date of the access token in memory
+  const endDate_access =
```

Comment:
1. wrong casing, we use camelCase in JS
2. unclear name, see the same comment regarding the refresh token 👆